src/main/java/magicbees/main/utils/compat/thaumcraft/NodeHelper.java

```diff
@@ -41,6 +44,9 @@ public static INode findNode(Chunk chunk, int x, int y, int z, int range) {
                 ((Map<ChunkPosition, TileEntity>) chunk.chunkTileEntityMap).values());
         Collections.shuffle(tileEntities);
         for (TileEntity entity : tileEntities) {
+            if (ThaumicHorizonsHelper.isActive() && entity instanceof TileSyntheticNode) {
+                return null;
+            }
```

Comment:
In case there's *real* node in the same chunk:
```suggestion
            if (ThaumicHorizonsHelper.isActive() && entity instanceof TileSyntheticNode) {
                continue;
            }
```

Reply:
Ironically it's been confirmed that this doesn't actually stop real nodes even if the shard's in the chunk

Reply:
Currently working on finding the other hole

Reply:
Probabaly, cause is that real node comes before shard in collection. If so, you should add another loop (before main one) which filters shards out.

Reply:
I'm gonna try basically putting a check in literally every instance where it calls a node in this class instead, getting someone who knows the dupe to try it for me now